bundles/org.eclipse.osgi/container/src/org/eclipse/osgi/internal/framework/EquinoxContainer.java

```diff
@@ -174,6 +179,13 @@ public EquinoxLogServices getLogServices() {
 	}
 
 	public Bundle getBundle(Class<?> clazz) {
+		if (connectModules.moduleConnector instanceof FrameworkUtilHelper) {
```

Comment:
We should need to do this if the helper is added to the `ConnectFrameworkUtilHelper` and plugged into `FrameworkUtil`.  Will this not lead to it being called twice?

Reply:
I think a Connect user will either register it as an SPI or through the creation directly.
The other idea was that we always give the chance first the one provided by the connect factory because I already found a little problem when running multiple frameworks that one has to find out what is the right one for that framework.

Reply:
My point is that I don't know why you call the ModuleConnector as a helper directly here when it should have been automatically added to the `ConnectFrameworkUtilHelper` and called below when we invoke `FrameworkUtil.getBundle`.

As for giving this helper a chance first.  I don't think we should do that.  Any helper has to be able to figure this stuff about the framework it is from itself because any code not using the deprecated `PackageAdmin` will instead directly call `FrameworkUtil.getBundle`.  But calling the same helper twice doesn't seem correct.